core/io/marshalls.cpp

```diff
@@ -1783,11 +1783,15 @@ Error encode_variant(const Variant &p_variant, uint8_t *r_buffer, int &r_len, bo
 				Variant variant = array.get_typed_script();
 				Ref<Script> script = variant;
 				if (script.is_valid()) {
-					String path = script->get_path();
-					ERR_FAIL_COND_V_MSG(path.is_empty() || !path.begins_with("res://"), ERR_UNAVAILABLE, "Failed to encode a path to a custom script for an array type.");
-					_encode_string(path, buf, r_len);
+					if (p_full_objects) {
+						String path = script->get_path();
+						ERR_FAIL_COND_V_MSG(path.is_empty() || !path.begins_with("res://"), ERR_UNAVAILABLE, "Failed to encode a path to a custom script for an array type.");
+						_encode_string(path, buf, r_len);
+					} else {
+						_encode_string(EncodedObjectAsID::get_class_static(), buf, r_len);
```

Comment:
`Array[EncodedObjectAsID]` may be incompatible with `Array[T]`.

Reply:
> `Array[EncodedObjectAsID]` may be incompatible with `Array[T]`.

Not sure I understand, this is exactly why when encoding with `p_full_objects = false` we **must** set the serialized array type to to `EncodedObjectAsID` since the content will actually be `EncodedObjectAsID`.

Reply:
This will cause an error when assigning the array to a typed variable.

Reply:
I've opened https://github.com/godotengine/godot/issues/94384 showing the other regression (fixed by this PR).